pyccel/parser/semantic.py

```diff
@@ -1364,13 +1364,13 @@ def _ensure_target(self, rhs, d_lhs):
             # TODO uncomment this line, to make rhs target for
             #      lists/tuples.
             rhs.is_target = True
-        if isinstance(rhs, IndexedElement) and rhs.rank > 0 and rhs.base.allocatable:
+        if isinstance(rhs, IndexedElement) and (rhs.rank > 0 and (rhs.base.allocatable or rhs.base.is_pointer)):
             d_lhs['allocatable'] = False
             d_lhs['is_pointer' ] = True
 
             # TODO uncomment this line, to make rhs target for
             #      lists/tuples.
-            rhs.base.is_target = True
+            rhs.base.is_target = True if rhs.base.allocatable else False
```

Comment:
What about stack arrays?

Reply:
i think i will change it to be True if rhs.base.is_pointer is false